src/Microsoft.ML.Data/Transforms/Hashing.cs

```diff
@@ -1211,15 +1211,15 @@ internal static bool IsColumnTypeValid(ColumnType type)
         /// </summary>
         /// <param name="env">Host Environment.</param>
         /// <param name="inputColumn">Name of the column to be transformed.</param>
-        /// <param name="outputColumn">Name of the output column. If this is null '<paramref name="inputColumn"/>' will be used.</param>
+        /// <param name="outputColumn">Name of the output column. </param>
         /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 31, inclusive.</param>
         /// <param name="invertHash">During hashing we constuct mappings between original values and the produced hash values.
         /// Text representation of original values are stored in the slot names of the  metadata for the new column.Hashing, as such, can map many initial values to one.
         /// <paramref name="invertHash"/> specifies the upper bound of the number of distinct input values mapping to a hash that should be retained.
         /// <value>0</value> does not retain any input values. <value>-1</value> retains all input values mapping to each hash.</param>
-        public HashingEstimator(IHostEnvironment env, string inputColumn, string outputColumn = null,
+        internal HashingEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
```

Comment:
>internal [](start = 8, length = 8)

why estimator constructor becomes internal?

Reply:
and shouldn't you swap output and input?

---
In reply to: [246971081](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246971081) [](ancestors = 246971081)

Reply:
Constructors become internal, so that users instantiate them only through the MlContext. See [https://github.com/dotnet/machinelearning/issues/2100](https://github.com/dotnet/machinelearning/issues/2100) 

---
In reply to: [246971081](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246971081) [](ancestors = 246971081)